### PR TITLE
SlackRTM: identity linking

### PR DESCRIPTION
**Pull request by OllieTerrance**
This allows a user, present in a SlackRTM-synced channel on both the Slack and Hangouts sides, to link their identities from each side.  The user must name their Google profile ID from within Slack, and name their Slack username from Hangouts, before the link the valid.

This allows Slack users to:
* run `/bot` commands as their Hangouts user, rather than the bot user
* trigger mention and keyword alerts for other Hangouts users

Limitation: you need the Slack config name in order to set a username on the Hangouts side of things.

### CODE REVIEW

**Comment by OllieTerrance** [comment]
I've been running this on my bot for the last week and it seems to work okay.  Not sure what else we might want triggered from Slack though -- it's a case of whitelisting features and calling their handlers from SlackRTM (I'm guessing there's no easy way to just trigger the message handler as that dupes the message in Slack).

**Comment by OllieTerrance** [comment]
Are there any thoughts about implementation of chatbridge profile sync yet?  Would be nice to get something similar to this working over there, though obviously don't want to port it in this style if there's a better way in the works...

**Comment by endofline** [comment]
Yes, we are planning a standardised profile sync across all external
bridges. We will take into account existing/suggested implementations like
telesync and this PR as well :) Although we cannot guarantee 1to1
compatibility, we will attempt to make the transition as straightforward as
possible.

File: hangupsbot/plugins/slackrtm.py [hunk 1]
```diff
@@ -684,6 +686,32 @@ def handleCommands(self, msg):
...
+            if not len(args):
+                ident = ""
+                message = u'%s: your identity has been cleared' % (msg.username)
+            elif not re.match('^[0-9]{21}$', args[0]):
+                message = u'%s: identity should be a Google profile ID, try using `hangoutmembers` to find yours' % (msg.username)
+            else:
+                ident = args[0]
+                message = u'%s: your identity has been set to %s' % (msg.username, ident)
+            if ident is not None:
+                idents = self.bot.user_memory_get(self.name, 'identities') or {'slack': {}, 'hangouts': {}}
```
Comment (endofline) [inline hangupsbot/plugins/slackrtm.py]: re: #797

**Comment by endofline** [comment]
keep this PR open so I can use it as a reference to implement the command sets and data structures required for identity linking in slackrtm for v3 chatbridges. will close manually after the featureset has been merged to that branch

**Comment by endofline** [comment]
merge to v3 features/chatbridges:
* moves `identify` to `plugins/slackrtm/commands_slack.py`
* moves `slack_identify` to `plugins/slackrtm/commands_hangouts.py`
* stores actual slack user ID instead of slack username as the identifier - probably not backward-compatible with your current keys in `memory.json`, @OllieTerrance 
* refactors code shared by both `identify` and `slack_identify`, and moved to `plugins/slackrtm/utils.py`
* removes the command bridging to be replaced by chatbridge version

the logic and usage for `identify`/`slack_identify` remains - for the most part - the same:

**link profiles**
1. `/bot slack_identify <slack team name> <slack user search>` on hangouts, where `<slack user search>` can be slack user id from `@hobot whoami` output or non-case sensitive slack username
2. `@hobot identify <hangouts user id>`, where `<hangouts user id>` is the hangouts "chat_id" from `/bot whoami`

note: running any of the above commands when a full mapping already exists will notify you whether you are able to remove the mapping or not - you cannot remove mappings that are "owned" by other users 

**delink profiles (changed)**
* on hangouts, `/bot slack_identify <slack team name> <slack user search> remove`
* on slack, `@hobot identify <hangouts user id> remove`

note: you only need to delink one-side of the profile to stop any profile sync features

**Comment by OllieTerrance** [comment]
LGTM 👍

No worries about compatibility --  I scripted all my mappings in the first time, easy enough to rerun with the new format.  🙂

**Comment by endofline** [comment]
https://github.com/hangoutsbot/hangoutsbot/pull/799/commits/c22a9ccf280ab38b73e64b18b4083d2d954cdc67 implements command bridging - any message from a synced user will go through the standard bot command handler with that user's context. this also has the added benefit of making the original message and hangouts user available to plugins such as subscribe and mentions

**Pull request merged**